### PR TITLE
NEW Elements can be deleted via the inline delete button. Fix versioning state in readBlocks query

### LINKED ISSUE #305: Implement delete button
The existing elemental editor (PHP GridField) has a delete button for quickly removing blocks.

The new React version needs this too. This will need to be written as a GraphQL mutation applied via an HOC.

## Acceptance criteria

- [x] Delete button shows on elements
- [x] Clicking the delete button removes the row
- [x] The element does not appear in the list on reload

## Pull requests

- [x] https://github.com/dnadesign/silverstripe-elemental/pull/314

### PR DESCRIPTION
**Pull request by robbieaverill**
This pull request adds the ability to delete an element from the editor using a similar button to that of the PHP GridField delete action.

Other changes:

* `.element-editor__element` is now a `span` rather than an anchor, since we're nesting a button inside it. It has the necessary `role` and `tabIndex` as well as click and keypress handlers for accessiblity.
* `ReadBlocksForPage` GraphQL query's versioning mode was incorrectly reading from the "latest" stage, it should be reading from draft since it's in the CMS

After the delete mutation runs it refetches the list of elements, which sets the `loading` prop on the `ElementList` to true while it's running. Once we have a loading indicator (#309) this will work automatically.

Resolves #305

### CODE REVIEW

**Comment by ScopeyNZ** [comment]
In this case do we have to bother with a refetch? Can't we just find a way to remove the element from the list and be done with it?

**Comment by robbieaverill** [comment]
That would involve mutating the blocks array prop. I think there is a way to do this with apollo's imperative cache API but I'm not sure how to do it yet - this is what we've been doing so far for history viewer anyway

File: client/src/components/ElementEditor/Header.js [hunk 1]
```diff
@@ -36,19 +63,39 @@ class Header extends Component {
           </div>
           <h3 className="element-editor-header__title">{title}</h3>
         </div>
+        <FormActionComponent
+          onClick={this.handleDelete}
+          title={null}
```
Comment (ScopeyNZ) [inline client/src/components/ElementEditor/Header.js]: This seems like a bug in admin 😒

**Comment by ScopeyNZ** [comment]
Do you think we should include some sort of loading indication during deletion before this is merged? Currently you are just staring at the block you deleted for a bit before it actually disappears.

**Comment by robbieaverill** [comment]
@ScopeyNZ yep, the loading prop is set by GraphQL but there's no indicator yet (noted in description)

**Comment by robbieaverill** [comment]
Merging with @ScopeyNZ's approval

File: client/src/components/ElementEditor/Header.js (hunk 1 above)
Comment (robbieaverill) [reply client/src/components/ElementEditor/Header.js]: To be fair, we could probably just omit the title prop

Comment (ScopeyNZ) [reply client/src/components/ElementEditor/Header.js]: You can - but you would also expect it to be passed through to the actual button that gets rendered...

Comment (ScopeyNZ) [reply client/src/components/ElementEditor/Header.js]: It already passed through a few attributes: https://github.com/silverstripe/silverstripe-admin/blob/1/client/src/components/FormAction/FormAction.js#L22-L26

**Pull request merged**